### DISCUSSION TITLE
Social preview image is missing for repository file link

### DISCUSSION THREAD

**Discussion by mieszko4** [community/community#88518] [Repositories]
### Select Topic Area

Bug

### Body

Hi,

#### Problem
We are trying to share https://github.com/rune/rune-multiplayer-web-games/blob/staging/open-source-grant/README.md on social media.
However, on some apps (e.g. X) it is missing the preview image.

#### Why it happens
This [test tool](https://www.opengraph.xyz/url/https%3A%2F%2Fgithub.com%2Frune%2Frune-multiplayer-web-games%2Fblob%2Fstaging%2Fopen-source-grant%2FREADME.md) shows the problem of missing image and title:
<img width="1727" alt="image" src="https://github.com/community/community/assets/3238370/99aaf7f4-e74f-44f6-a248-3c9ae481f91b">

The reason for this seems to be that json is returned on default (i.e. when `Accept: text/html` header is missing), which causes that some social apps are not seeing html when parsing social meta tags. This can be tested using curl:
```sh
curl https://github.com/rune/rune-multiplayer-web-games/blob/staging/open-source-grant/README.md

{"payload":{"allShortcutsEnabled":false,"fileTree"....
```

On the other hand, the main repo page (https://github.com/rune/rune-multiplayer-web-games) [shows correctly](https://www.opengraph.xyz/url/https%3A%2F%2Fgithub.com%2Frune%2Frune-multiplayer-web-games) because it returns html on default.

#### Questions
1. Should GitHub return html on default?
2. Is there some query parameter that will tell GitHub to return html instead of json?
3. Any other suggestions how to resolve this issue?

Thnx!
Miłosz

**Comment by SilasRodrigues19** [comment]
Hey @mieszko4 

The social preview appears when you share your repository's base URL

(https://github.com/rune/rune-multiplayer-web-games)

And normally when you replace the image, it takes a while to be recognized on some social networks.

But notice that in X/Twiiter, when the base URL of the repository is shared, the social preview appears:

![image](https://github.com/community/community/assets/47532703/bf451cd5-18e7-4711-b909-556dda6109e6)

**Reply by mieszko4** [reply]
Thnx @SilasRodrigues19,

I am aware that repository's base url works.
My issue is specifically with sharing subpages (i.e. https://github.com/rune/rune-multiplayer-web-games/blob/staging/open-source-grant/README.md)

**Reply by SilasRodrigues19** [reply]
I understand @mieszko4, but I don't think this is a bug per se. The social preview is not displayed on subpages, I think you would need to configure the og image on your website as follows:

```html
<meta property="og:title" content={title} />
<meta property="og:description" content={description} />
<meta property="og:url" content={canonicalURL} />

<!-- Here the image that will be displayed when the website is shared on social networks will be configured. -->
<meta property="og:image" content={socialImageURL} />
```

**Reply by mieszko4** [reply]
Thnx.

I was expecting a consistent behavior when sharing repository's base URL and repository's subpages hence I think it is a bug.
I could use meta tags, however, that would require setting up GitHub Pages which I wanted to avoid.

Also note that meta tags are already there if I request the subpage with `Accept: text/html` so subpages are ready for social preview. The problem is that Twitter does not seem to use `Accept: text/html` when scraping the page for meta tags.

So to fix the problem, either GitHub needs change the default to return html (instead of json) or Twitter (and all other social pages) needs to always add `Accept: text/html` header.
